src/components/Notification.tsx

```diff
@@ -0,0 +1,36 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { NotificationType } from '../types/NotificationType';
+import { useGlobalContext } from '../hooks/useGlobalContext';
+
+type Props = {
+  type: NotificationType;
+} & (
+  {
+    type: NotificationType.PostsWarning;
+    warningText: string
+  } | {
+    type: Exclude<NotificationType, NotificationType.PostsWarning>;
+    warningText?: string;
+  }
+);
+
+export const Notification: FC<Props> = ({ type, warningText }) => {
```

Comment:
The 'warningText' prop is used regardless of the NotificationType. However, it is declared optional for types other than 'PostsWarning'. It would be clearer to always require this prop and ensure that a meaningful value is passed for all notification types.